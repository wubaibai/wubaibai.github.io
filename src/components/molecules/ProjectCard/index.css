.card {
	color: white;
}

.cover {
	background: rgba(255, 255, 255, 0.1);
	margin-bottom: 16px;
	line-height: 0;
	border-radius: 4px;
	overflow: hidden;

	& > img {
		width: 100%;
	}
}

.placeholder {
	background: rgba(255, 255, 255, 0.1);
	line-height: 0;
	border-radius: 4px;
	overflow: hidden;
	height: 180px;
	display: flex;
	align-items: center;
	justify-content: center;

	& > * {
		margin-right: 20px;
	}

	& > *:last-child {
		margin-right: 0;
	}
}

.title {
	margin-bottom: 12px;
    font-family: var(--fontNarrow);
    font-size: 20px;
    font-weight: 900;
}

.description {
	margin-bottom: 20px;
	font-size: 14px;
	line-height: 1.4;
	letter-spacing: 1px;
	font-weight: 300;
}

.tags {
	align-self: flex-end;
}