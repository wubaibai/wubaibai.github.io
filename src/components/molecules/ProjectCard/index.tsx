/* eslint-disable react/jsx-no-target-blank */
import React from 'react';
import classNames from 'classnames';

import Tags from 'components/molecules/Tags';
import Dots from 'components/molecules/Dots';

import style from './index.css';

interface ProjectProperty {
	title: string;
	description?: string;
	cover: string;
	tags: string[];
	link?: string;
}

interface ProjectCardProperty {
	data?: ProjectProperty;
	className?: string;
}

const ProjectCard: React.FC<ProjectCardProperty> = ({ data, className }) => {
	if (!data) {
		return (
			<div className={classNames(style.card, className)}>
				<div className={style.placeholder}>
					<Dots count={3} />
				</div>
			</div>
		);
	}

	return (
		<div className={classNames(style.card, className)}>
			<div className={style.cover}>
				{data.link && (
					<div className={style.overlay}>
						<a className={style['overlay-link']} href={data.link} target="_blank">
							<div className={style['overlay-button']}>
								<div className={style['overlay-button-text']}>去瞧瞧</div>
								<div className={style['overlay-button-base']}> </div>
							</div>
						</a>
					</div>
				)}
				<img src={data.cover} alt={data.title} />
			</div>
			<div className={style.title}>{data.title}</div>
			<div className={style.description}>{data.description}</div>
			<div className={style.tags}>
				<Tags items={data.tags} />
			</div>
		</div>
	);
};

export default ProjectCard;
