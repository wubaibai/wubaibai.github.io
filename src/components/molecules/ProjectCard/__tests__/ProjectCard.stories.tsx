import React from 'react';
import { storiesOf } from '@storybook/react';

import ProjectCard from 'components/molecules/ProjectCard';
import coverAwesomeShare from 'images/cover/awesome-share.png';

const stories = storiesOf('molecules/ProjectCard', module);

const project = {
	cover: coverAwesomeShare,
	title: 'Awesome Share / 連接分享設定工具',
	description:
		'透過 Firebase 打造網址分享小工具，製作一個網頁可以自己上傳圖片、連結、Meta 內容，儲存於 Firebase 產生分享連結，可以在同一頁面撰寫 PO 文，一鍵分享至 Facebook。',
	tags: ['pug', 'firebase'],
};

stories.add('__interactive', () => <ProjectCard data={project} />, { jest: 'ProjectCard' });
