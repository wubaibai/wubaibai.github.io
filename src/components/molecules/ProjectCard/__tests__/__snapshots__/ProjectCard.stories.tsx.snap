// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="cover"
    >
      <img
        alt="Awesome Share / 連接分享設定工具"
        src="test-file-stub"
      />
    </div>
    <div
      class="title"
    >
      Awesome Share / 連接分享設定工具
    </div>
    <div
      class="description"
    >
      透過 Firebase 打造網址分享小工具，製作一個網頁可以自己上傳圖片、連結、Meta 內容，儲存於 Firebase 產生分享連結，可以在同一頁面撰寫 PO 文，一鍵分享至 Facebook。
    </div>
    <div
      class="tags"
    >
      <div
        class="tags"
      >
        <div
          class="tag"
        >
          <div
            class="text"
          >
            pug
          </div>
          <div
            class="base"
          />
        </div>
        <div
          class="tag"
        >
          <div
            class="text"
          >
            firebase
          </div>
          <div
            class="base"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
